src/words/environment/WordsObject.java

```diff
@@ -13,13 +15,35 @@
 	private Position cell;
 	private String currentMessage;
 	private Action lastAction;
-
+	private HashMap<WordsObject, ArrayList<Property>> referers;
```

Comment:
Sorry, I was referring specifically to the ArrayList part, not the entire thing.  An ArrayList provides more functionality than we need to store the Propertys, since there's no need to retrieve them by index or iterate through them in a specific order.  But it's certainly OK as is. :-)